sdk/src/account.rs

```diff
@@ -40,6 +41,142 @@ pub struct AccountSharedData {
     pub rent_epoch: Epoch,
 }
 
+impl<'de> Deserialize<'de> for AccountSharedData {
```

Comment:
This is not desirable.

Reply:
I found a simpler way to implement serialize/deserialize with minimal performance impact and not a lot of new code.

Reply:
Does it need to be serialize/deserialize? Maybe it makes sense for it to be in another module and then that module excluded from contract code.

It can be in this list as `account_shared_data` module for instance:
```
default = [
  "full" # functionality that is not compatible or needed for on-chain programs
]
full = [
    "assert_matches",
    "byteorder",
    ```

Reply:
This might need some debugging what is going on. We have examples of `Serialize/Deserialize` with inner `Arc` fields which work fine:
https://github.com/solana-labs/solana/blob/ff2242d0e/runtime/src/epoch_stakes.rs#L15-L21

Reply:
I think it's compiling in the BPF toolchain environment. Because Account is used for both, so everything in this module would get compiled for both. That environment has more limitations than the normal native dev environment.

Reply:
If we don't need something in the BPF environment, we should probably move it out if we can.

Reply:
@behzadnouri 
[here's a failing build:](https://buildkite.com/solana-labs/solana/builds/42372#cf928dd8-25a9-4a5c-9500-696e3e4a7d45)

Reply:
I will try a different approach where we separate it out. it looks like it doesn't need to serialize.

Reply:
I have removed serialization completely. Moving out AccountSharedData is a different issue. Yay for the pr process!

Reply:
Could not make AccountSharedData unavailable in bpf due to KeyedAccounts.